staging/src/k8s.io/legacy-cloud-providers/gce/gce_instances.go

```diff
@@ -83,29 +83,81 @@ func (g *Cloud) ToInstanceReferences(zone string, instanceNames []string) (refs
 }
 
 // NodeAddresses is an implementation of Instances.NodeAddresses.
-func (g *Cloud) NodeAddresses(_ context.Context, _ types.NodeName) ([]v1.NodeAddress, error) {
-	internalIP, err := metadata.Get("instance/network-interfaces/0/ip")
-	if err != nil {
-		return nil, fmt.Errorf("couldn't get internal IP: %v", err)
+func (g *Cloud) NodeAddresses(ctx context.Context, nodeName types.NodeName) ([]v1.NodeAddress, error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, 1*time.Hour)
+	defer cancel()
+
+	instanceName := string(nodeName)
+
+	if g.useMetadataServer {
+		// Use metadata server if possible
+		if g.isCurrentInstance(instanceName) {
+
+			nics, err := metadata.Get("instance/network-interfaces/")
+			if err != nil {
+				return nil, fmt.Errorf("couldn't get network interfaces: %v", err)
+			}
+
+			nicsArr := strings.Split(nics, "/\n")
+			nodeAddresses := []v1.NodeAddress{}
+
+			for _, nic := range nicsArr {
+
+				if nic == "" {
+					continue
+				}
+
+				internalIP, err := metadata.Get(fmt.Sprintf("instance/network-interfaces/%v/ip", nic))
```

Comment:
It would be better if these strings are extracted as constants. Also, use `%s` instead of `%v` for less confusion. Example: https://github.com/kubernetes/kubernetes/blob/v1.18.2/staging/src/k8s.io/legacy-cloud-providers/gce/gce_disks.go#L55

Reply:
Can we not conflate the multi-nic on instance change with the getting addresses remotely change? Not sure what effect returning multiple addresses to Kubelet will cause and would rather the fix for Cloud Controller Manager not be rolled back as a result.

Reply:
So how do we fix #90570 ?

Reply:
Easiest way would be...In another PR?

Reply:
Okay I'm removing the 2nd commit from this PR and just keeping the single interface code